### PR TITLE
Adding CLientHTTPTimeoutError for Timeout issues

### PR DESCRIPTION
**Pull request by abhinab22mohanty**
Adding Standardized errors for timeout issue scenarios
ClientHTTPTimeoutError: Server took too long to respond

image

Improvement on PR : https://github.com/microsoft/omnichannel-sdk/pull/61

### CODE REVIEW

File: src/SDK.ts [hunk 1]
```diff
@@ -202,6 +202,9 @@ export default class SDK implements ISDK {
       } catch (error) {
         const elapsedTimeInMilliseconds = timer.milliSecondsElapsed;
         this.logWithLogger(LogLevel.ERROR, OCSDKTelemetryEvent.GETLWISTATUSFAILED, "Get LWI Details failed", requestId, undefined, elapsedTimeInMilliseconds, requestPath, method, error);
+        if (error.code === "ECONNABORTED") {
+          throw new Error(SDKError.ClientHTTPTimeoutErrorName + ":" + SDKError.ClientHTTPTimeoutErrorMessage);
```
Comment (xTEddie) [inline src/SDK.ts:206]: Can we use string interpolation?

const message = \`${SDKError.ClientHTTPTimeoutErrorName}: ${SDKError.ClientHTTPTimeoutErrorMessage}\`

Comment (xTEddie) [reply src/SDK.ts:206]: Should we have a util `throwClientHTTPTimeoutError`?

```
const throwClientHTTPTimeoutError = () => {
  const message = `${SDKError.ClientHTTPTimeoutErrorName}: ${SDKError.ClientHTTPTimeoutErrorMessage}`
  throw new Error(message);
}
```

File: src/SDK.ts [hunk 2]
```diff
@@ -297,7 +301,11 @@ export default class SDK implements ISDK {
 
       // Base case
       if (currentRetryCount + 1 >= this.configuration.getChatTokenRetryCount) {
-        reject(new Error(`The retry for getchattoken has exceeded its max value of ${this.configuration.getChatTokenRetryCount}`));
+        if (getChatTokenError && getChatTokenError.code == "ECONNABORTED") {
```
Comment (xTEddie) [inline src/SDK.ts:304]: I think an `if` is good enough. A `throw` would end the code execution. 

```
if (getChatTokenError && getChatTokenError.code == "ECONNABORTED") {
  throw new Error(); // Throw error & end code execution
}

reject(new Error()); // Method not excute
```

Comment (xTEddie) [reply src/SDK.ts:304]: Should `ECONNABORTED` be an `axiosConstant`?

**Review by xTEddie — APPROVED** [review]

File: src/SDK.ts [hunk 3]
```diff
@@ -938,7 +969,9 @@ export default class SDK implements ISDK {
       } catch (error) {
         const elapsedTimeInMilliseconds = timer.milliSecondsElapsed;
         this.logWithLogger(LogLevel.ERROR, OCSDKTelemetryEvent.FETCHDATAMASKINGFAILED, "Fetch Data Masking Failed", requestId, undefined, elapsedTimeInMilliseconds, requestPath, method, error);
-
+        if (error.code === Constants.axiosTimeoutErrorCode) {
+          throwClientHTTPTimeoutError();
+        }
```
Comment (elopezanaya) [inline src/SDK.ts:974]: this 3 lines are repeated a lot of times, if in the future we need to modify it, we willneed to go over this in all the places , with the risk of missing one, 

could we use a function and just invoke it?, so for future modification will be in a single place, 

DRY.

**Review by elopezanaya — CHANGES_REQUESTED** [review]

**Pull request merged**